### PR TITLE
Issue #4052: update config of SingleLineJavadoc in google_checks.xml

### LINKED ISSUE #4052: SingleLineJavadoc update for new requirements what tags are allowed in singleline javadoc
parent issue - #3888 

guide update - https://github.com/google/styleguide/commit/594d91bfbb8dfcbcac0f148af8175f2b7ec9857b#diff-b6c9191ee5ccb7a5ef170e98e51d94d8L1084

> The basic form is always acceptable. The single-line form may be substituted when the entirety
of the Javadoc block (including comment markers) can fit on a single line. Note that this only
applies when there are no block tags such as `@return`.

http://checkstyle.sourceforge.net/config_javadoc.html#SingleLineJavadoc

we already have property - `ignoredTags` and it is null by default so no block tags are allowed.
`ignoreInlineTags` - is true by default.

We need to change our google config to ignore inline tags (switch to default by removing `ignoreInlineTags`).
current code:
https://github.com/checkstyle/checkstyle/blob/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml#L343-L345

definition of "block tags" (exact list of them) - http://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDBEFIF ([for java7](http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#javadoctags))
Lets save this link in xdoc to explain user meaning of such terms.

lets do updates for our IT inputs, case from #3762 should be added too.

----------

**Current behavior:**
```
/var/tmp$ cat Test.java 
public class Test {
  /** Create a new {@code Test} instance. */
  public Test() {}
}

/var/tmp$ java -jar checkstyle-8.36.2-all.jar -c /google_checks.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:2: Single-line Javadoc comment should be multi-line. [SingleLineJavadoc]
Audit done.
```

**Expected:** no violations.

### PR DESCRIPTION
**Pull request by himanshurijal**
Resolves #4052

### CODE REVIEW

**Comment by himanshurijal** [comment]
@romani I tried to update test and config files following your discussion in #4052. Please let me know if any changes are required.

**Review by strkkk — APPROVED** [review]

**Review by romani — CHANGES_REQUESTED** [review]
from issue:
> definition of "block tags" (exact list of them) - http://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDBEFIF (for java7)
Lets save this link in xdoc to explain user meaning of such terms.

please update xdoc 
https://github.com/checkstyle/checkstyle/blob/9c53648515ba2208ca50c7078fc059328424ee3e/src/xdocs/config_javadoc.xml#L3314
and at
https://github.com/checkstyle/checkstyle/blob/9c53648515ba2208ca50c7078fc059328424ee3e/src/xdocs/config_javadoc.xml#L3321
with link to oracle web site from above 

link should be at table - https://checkstyle.org/config_javadoc.html#SingleLineJavadoc_Properties

**Comment by himanshurijal** [comment]
I tried to update link as you specified. However, I have two failing tests now. 
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=1237&view=logs&j=e663805c-ee13-5a24-f979-d725e0346c8e&t=40ff6e43-c5d7-5753-f947-23012624b0c6&l=943
I know how to fix the error with  XdocsJavaDocsTest (sync Javadoc with XML file), but I am having trouble understanding the failure with XdocsPagesTest.

**Comment by himanshurijal** [comment]
> please update xdoc

I tried to update the link as you specified. However, I have two failing tests now.
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=1237&view=logs&j=e663805c-ee13-5a24-f979-d725e0346c8e&t=40ff6e43-c5d7-5753-f947-23012624b0c6&l=943
I know how to fix the error with XdocsJavaDocsTest (sync Javadoc with XML file), but I am having trouble understanding the failure with XdocsPagesTest.

File: src/xdocs/config_javadoc.xml [hunk 1]
```diff
@@ -3311,14 +3311,20 @@ public boolean testMethod(int firstParam) {
               <td>8.3</td>
             </tr>
             <tr>
-              <td>ignoredTags</td>
+              <td>
+                <a href = "http://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDBEFIF">
+                ignoredTags</a>
+              </td>
               <td>Specify block tags which are ignored by the check.</td>
```
Comment (romani) [inline src/xdocs/config_javadoc.xml:3318]: please make a link over `block tags` text in description. The same for other property.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/xdocs/config_javadoc.xml (hunk 1 above)
Comment (himanshurijal) [reply src/xdocs/config_javadoc.xml]: Done.

I tried doing this before. While, XdocsPagesTest is not failing anymore. I still get an error with XdocsJavaDocsTest.

https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=1249&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=296

Screen Shot 2021-02-19 at 9 34 31 AM

I don't understand why `Actual   :Specify block tags which are ignored by the check.` is shown in the test. Even though I have included the link in the source code for SingleLineJavadocCheck.java file.

Comment (romani) [reply src/xdocs/config_javadoc.xml]: Please try to update text over setter.

**Comment by romani** [comment]
@himanshurijal , are you stuck with problem ?
if not please find time to fix CIs failures

File: src/xdocs/config_javadoc.xml (hunk 1 above)
Comment (himanshurijal) [reply src/xdocs/config_javadoc.xml]: Sorry, I did not understand what you meant by "update text over setter".

**Comment by himanshurijal** [comment]
@romani I tried to make a number of different changes, but all led to the same error with the test.

If xdoc and javadoc are synched, the test reads the wrong input from SingleLineJavadocCheck.java for `Actual` string as I showed above.

If xdoc and javadoc are not synched, the test correctly reads input SingleLineJavadocCheck.java for `Actual` string (but complains that xdoc and javadoc are not synched which is true).

Screen Shot 2021-02-22 at 4 52 03 PM

I tried debugging but I found the test was too complex (it has too many calls to different methods) for me to debug with my current understanding of the source code. At this point, I am not sure what to do. Should I close PR and work on some other issue?

**Comment by himanshurijal** [comment]
@romani I tried to make a number of different changes, but all led to the same error with the test.

If xdoc and javadoc are synched, the test reads the wrong input from SingleLineJavadocCheck.java for `Actual` string as I showed above.

If xdoc and javadoc are not synched, the test correctly reads input SingleLineJavadocCheck.java for `Actual` string (but complains that xdoc and javadoc are not synched which is true).

Screen Shot 2021-02-22 at 4 52 03 PM


I tried debugging but I found the test was too complex (it has too many calls to different methods) for me to debug with my current understanding of the source code. At this point, I am not sure what to do. Should I close PR and work on some other issue?

File: src/xdocs/config_javadoc.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/config_javadoc.xml]: Javadoc over setter method for this field.
`setIgnoredTags"

**Comment by romani** [comment]
You can try another issue, while I am waiting for laptop time to look closely your problem.

File: src/xdocs/config_javadoc.xml (hunk 1 above)
Comment (himanshurijal) [reply src/xdocs/config_javadoc.xml]: Thank you for the clarification!
I also had to update the text over variable declaration along with the setter to get the test passing.

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**